### DISCUSSION TITLE
react-router-dom-v5-compat breaks v5 typings

### DISCUSSION THREAD

**Discussion by ghost** [remix-run/react-router#9892] [v5 to v6 Migration]
Hey folks,
After I installed `react-router-dom-v5-compat` I got a lot of type errors.
I assume it's because `@types/react-router-dom` exports many types from `react-router`, and `react-router` is at v6.6.2.
Before, because `react-router` did not have types, it would resolve them from `@types/react-router` but now it defaults to the library.

Any way to get around this?

**Comment by mgmarlow** [comment]
Late to the party here but I just copied the @types/react-router and @types/react-router-dom declarations into my own custom file, `my-project/@types/react-router-dom.d.ts`. The I removed the definitely-typed package from my project.

I also needed to add an import `@types/history` as well.

**Comment by andreasgerstmayr** [comment]
In my case, adding `@types/react-router` explicitly to the `devDependencies` in `packages.json` resolved the issue.
When only `@types/react-router-dom` was declared, I got type issues because typings got resolved to the types declared in `@remix-run/router`.
